sdk/core/Azure.Core/src/Internal/RehydrationOperationOfT.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System.ClientModel.Primitives;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+#pragma warning disable SA1649 // File name should match first type name
+    internal class RehydrationOperation<T> : Operation<T> where T : IPersistableModel<T>
+#pragma warning restore SA1649 // File name should match first type name
+    {
+        private readonly OperationInternal<T> _operation;
+        private readonly NextLinkOperationImplementation _nextLinkOperation;
+
+        public RehydrationOperation(HttpPipeline pipeline, RehydrationToken? rehydrationToken, ClientOptions options = null)
+        {
+            Argument.AssertNotNull(pipeline, nameof(pipeline));
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+
+            IOperationSource<T> source = new GenericOperationSource<T>();
+            _nextLinkOperation = (NextLinkOperationImplementation)NextLinkOperationImplementation.Create(pipeline, rehydrationToken);
+            var operation = NextLinkOperationImplementation.Create(source, _nextLinkOperation);
+            var operationState = operation.UpdateStateAsync(false, default).EnsureCompleted();
+            _operation = operationState.HasCompleted
+                    ? new OperationInternal<T>(operationState)
+                    :  new OperationInternal<T>(operation, new ClientDiagnostics(options ?? ClientOptions.Default), operationState.RawResponse);
+        }
+
+        public override T Value => _operation.Value;
+
+        public override bool HasValue => _operation.HasValue;
+
+        public override string Id => _nextLinkOperation.OperationId ?? null;
```

Comment:
Same comment about not violating the nullability contract as above.

Reply:
removed nullable